Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM golang:1.21 AS builder
+
+WORKDIR /building
+COPY . .
+
+ARG APP
+RUN make ${APP}
+
+FROM alpine:3.18 AS runtime
+
+ARG APP
+RUN addgroup -g 1000 -S ${APP} && adduser -u 1000 -S ${APP} -G ${APP} --home /app \
+ && echo -e "#!/bin/sh\nexec /usr/local/bin/${APP} \$@" > /app/entrypoint.sh \
```

Comment:
What's the purpose of this script? What are the differences compared to the previous execution method?

Reply:
The reason for introducing this script was the unification of the 2 Dockerfiles into 1.
It's not allowed to use ARG/ENV variables inside ENTRYPOINT, your binaries have different names, it's best practice to create a minimal `exec` script in this case.
I will improve it a bit to make hadolint happy 😄